MailCore/Utils/SentryDebug.swift

```diff
@@ -343,3 +344,64 @@ public enum SentryDebug {
         return true
     }
 }
+
+// MARK: - SHARED -
+
+public extension SentryDebug {
+    static func addBreadcrumb(
+        message: String,
+        category: SentryDebug.Category,
+        level: SentryLevel,
+        metadata: [String: Any]? = nil
+    ) {
+        Task {
+            let breadcrumb = Breadcrumb(level: level, category: category.rawValue)
+            breadcrumb.message = message
+            breadcrumb.data = metadata
+            SentrySDK.addBreadcrumb(breadcrumb)
+        }
+    }
+
+    static func capture(
+        error: Error,
+        context: [String: Any]? = nil,
+        contextKey: String? = nil,
+        extras: [String: Any]? = nil
+    ) {
+        Task {
+            SentrySDK.capture(error: error) { scope in
+                if let context, let contextKey {
+                    scope.setContext(value: context, key: contextKey)
+                }
+
+                if let extras {
+                    scope.setExtras(extras)
+                }
+            }
+        }
+    }
+
+    static func capture(
+        message: String,
+        context: [String: Any]? = nil,
+        contextKey: String? = nil,
+        level: SentryLevel? = nil,
+        extras: [String: Any]? = nil
+    ) {
+        Task {
```

Comment:
We have to be careful about what we send as `context`. If it's a realm object it will crash

Reply:
I merged this code with `addAsyncBreadcrumb`, and made explicit the async nature of the new methods.